js/src/instructions/item.ts

```diff
@@ -0,0 +1,1227 @@
+          )[0],
+          craftItemClass: (
+            await getItemPDA(args.craftItemClassMint, args.craftItemClassIndex)
+          )[0],
+          receiver: (this.program.client.provider as AnchorProvider).wallet
+            .publicKey,
+          systemProgram: SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+        })
+        .remainingAccounts(remainingAccounts)
+        .instruction(),
+    ];
+  }
+
+  async beginItemActivation(
+    args: BeginItemActivationArgs,
+    accounts: BeginItemActivationAccounts,
+    _additionalArgs: BeginItemActivationAdditionalArgs = {}
+  ) {
+    const remainingAccounts =
+      await generateRemainingAccountsGivenPermissivenessToUse({
+        permissivenessToUse: args.usagePermissivenessToUse,
+        tokenMint: accounts.itemMint,
+        parentMint: args.itemClassMint,
+        parentIndex: args.classIndex,
+        parent: (await getItemPDA(args.itemClassMint, args.classIndex))[0],
+        metadataUpdateAuthority: accounts.metadataUpdateAuthority,
+        program: this.program.client,
+      });
+
+    const itemClassKey = (
+      await getItemPDA(args.itemClassMint, args.classIndex)
+    )[0];
+
+    const [itemActivationMarker, itemActivationBump] =
+      await getItemActivationMarker({
+        itemMint: accounts.itemMint,
+        index: args.index,
+        usageIndex: new BN(args.usageIndex),
+        amount: args.amount,
+      });
+
+    const instructions: TransactionInstruction[] = [];
+    const itemTransferAuthority = accounts.itemTransferAuthority;
+
+    if (
+      accounts.itemAccount &&
+      accounts.itemAccount.equals(
+        (
+          await getAtaForMint(
+            accounts.itemMint,
+            (this.program.client.provider as AnchorProvider).wallet.publicKey
+          )
+        )[0]
+      )
+    ) {
+      instructions.push(
+        Token.createApproveInstruction(
+          TOKEN_PROGRAM_ID,
+          accounts.itemAccount,
+          itemTransferAuthority.publicKey, // QUESTION: itemTransferAuthority could be null - what to do then?
+          (this.program.client.provider as AnchorProvider).wallet.publicKey,
+          [],
+          args.amount.toNumber()
+        )
+      );
+    }
+
+    const itemKey = (await getItemPDA(accounts.itemMint, args.index))[0];
+
+    const validationKey =
+      args.itemClass.object.itemClassData.config.usages?.[args.usageIndex]
+        .validation?.key;
+    const validationProgram: PublicKey = !!validationKey
+      ? new web3.PublicKey(validationKey)
+      : SystemProgram.programId;
+
+    instructions.push(
+      await this.program.client.methods
+        .beginItemActivation(args)
+        .accounts({
+          itemClass: itemClassKey,
+          itemMint: accounts.itemMint,
+          item: itemKey,
+          itemAccount: accounts.itemAccount as Address,
+          itemTransferAuthority: itemTransferAuthority.publicKey, // TODO: Update based on above
```

Comment:
If it's null, don't add it to the accounts object. Since we will be throwing above if we need itemTransferAuthority, we are handling it not existing already and can just assign it based on if it exists or not.